backend/sozisel/lib/sozisel_web/schema/resolvers/presenter_resolvers.ex

```diff
@@ -0,0 +1,89 @@
+defmodule SoziselWeb.Schema.Resolvers.PresenterResolvers do
+  alias SoziselWeb.Schema.Helpers
+  alias SoziselWeb.Schema.Subscriptions.Topics
+
+  alias Sozisel.Model.{
+    Sessions,
+    Events,
+    LaunchedEvents,
+    Quizzes
+  }
+
+  alias Events.Event
+  alias LaunchedEvents.LaunchedEvent
+  alias Sessions.Session
+  alias Quizzes.Quiz
+
+  def prepare_data_for_participants(
+        %Event{} = event,
+        %LaunchedEvent{} = launched_event
+      ) do
+    event_data = event.event_data
+
+    participant_event_data =
+      case event_data.__struct__ do
+        Quiz -> Events.prepare_quiz_data_for_participants(event_data)
+        _ -> {:error, "Wrong data"}
+      end
+
+    %{
+      id: launched_event.id,
+      name: event.name,
+      event_data: participant_event_data
+    }
+  end
+
+  def send_event_to_all(_parent, %{event_id: event_id, session_id: session_id}, _ctx) do
```

Comment:
As mentioned above, we should probably have just a one resolver (you could do the pattern matching on `broadcast: true` so same resolver but 2 definitions).

Reply:
And I agree that for those events you can add another topic that will broadcast to the room but only for participants so the topic could be `session:#{session_id}:participants` or something like this.